BJS2-40520 Организация события номер

Darkselect opened the pull request:
Сделал первые 9 пунктов. 1/7

Darkselect committed:
Организация события номер задача: BJS2-40520

Darkselect committed:
BJS2-40520: Восстановил структуру папок

DeemaP commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,49 @@
+
+@Component
+@RequiredArgsConstructor
+public class EventService {
```
Привет! В видеолекции про MapStruct на 45-й минуте примерно говорится о том, что один сервис должен работать с одним репозиторием, а у тебя тут два репозитория. По-хорошему тебе нужно с UserRepository работать через UserService)

DeemaP commented on src/test/java/school/faang/user_service/service/EventServiceTest.java:
```diff
@@ -0,0 +1,132 @@
+import school.faang.user_service.service.event.EventService;
+import school.faang.user_service.validation.EventValidation;
+
+import java.util.*;
```
Насколько знаю, такие импорты со звездочкой - моветон)
Поставь в идее значение wildcard imports 9999
File > Settings > Editor > Code Style > Java > Imports >
Class count to use import with ‘*’: 9999
Names count to use static import with ‘*’: 9999


![Screenshot 2024-10-31 at 21 37 20](https://github.com/user-attachments/assets/3335ac7c-8f8b-4e12-baac-652cae39f6c8)

DeemaP reviewed (commented):
Привет! Оставил пару комментариев, где уверен, что нужно кое что поправить)
И еще, кажется все пока не создавали пакеты внутри общих пакетов service, master  и тд для сервисов, мапперов и дтошек. Просто создавали соответствующие классы прям в первом пакете, думаю стоит всем обсудить этот нюанс, а то при первом мерже получится, что у всех сервисы, маппера и дто лежат прям в пакетах, а у тебя пакеты в пакетах (СОРИ, жестко загнул, но надеюсь мысль донес) 

Хорошего вечера!!!

drewbal commented on src/main/java/school/faang/user_service/mapper/EventMapper.java:
```diff
@@ -0,0 +1,12 @@
+@Mapper(componentModel = "spring", unmappedTargetPolicy = org.mapstruct.ReportingPolicy.IGNORE)
+public interface EventMapper {
+    EventDto eventToDto(Event event);
+    Event dtoToEvent(EventDto eventDto);
```
У тебя в EventoDto есть поле Long OwnerId. При маппинге ты потрешь в объекте все поля этой сущности. Надо делать аннотацию с ignore. Это же относится к списку List<SkillDto> relatedSkills, если названия полей сущности и DTO совпадают

drewbal commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,49 @@
+        User user = userRepository.findById(event.getOwnerId()).orElseThrow(() -> new NoSuchElementException("User not found"));
+        List<Skill> userOwnerSkills = user.getSkills();
+
+        Event findEvent = eventRepository.findById(event.getId()).orElseThrow(() -> new NoSuchElementException("Event not found"));
```
А вот если создать отдельный маленький сервисный метод типа findEventById(Long id), то можно сократить тут количество кода, убрав проверку и выбрасывание исключения в метод, а также во всех прочих случаях, когда может понадобиться такая простая и нужная операция как поиск ивента по его id :)

Аналогичная мысль касается и поиска юзера в строке 29, но для этого надо пили UserService. К слову, я его сделал (и думаю не только я), так что потом можно подрихтовать код

Darkselect replied:
Принято. Работаю

drewbal commented on src/main/java/school/faang/user_service/validation/EventValidation.java:
```diff
@@ -0,0 +1,18 @@
+public class EventValidation {
+    public void validateEvent(EventDto event) {
+        if (event.getTitle().isBlank()) {
+            throw new DataValidationException("Empty or null title");
```
а isBlank включает в себя Null или только пустые строки (которые еще не null)?

Darkselect replied:
Невнимательность. Добавил проверку на null

drewbal reviewed (commented):
случайно нажал на ревью

Darkselect committed:
BJS2-40520: Полностью закончил задачу 7/7

А конкертно:
1.Создать событие
2.Получить событие
3.Получить все события с фильтрами
4.Удалить событие
5.Обновить событие
6.Получить все созданные пользователем события
7.Получить все события, в которых пользователь принимает участие

drewbal commented on src/main/java/school/faang/user_service/controller/EventController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public void getOwnedEvents(long userId) {
+        eventService.getOwnedEvents(userId);
```
У тебя метод буквально говорит, что получает собственные ивенты, но при этом ничего не возвращает. Как так может быть? Наверное надо вернуть List<EventDto>

drewbal commented on src/main/java/school/faang/user_service/controller/EventController.java:
```diff
@@ -0,0 +1,48 @@
+        eventService.getOwnedEvents(userId);
+    }
+
+    public void getParticipatedEvents(long userId) {
```
Аналогично

drewbal commented on src/main/java/school/faang/user_service/mapper/EventMapper.java:
```diff
@@ -0,0 +1,19 @@
+public interface EventMapper {
+    @Mapping(target = "ownerId", ignore = true)
+    @Mapping(target = "relatedSkills", ignore = true)
+    EventDto eventToDto(Event event);
```
А вот когда ты из сущности в дто переводишь, то не надо бояться ничего перетереть. Тут ты должен сделать маппинг поля сущности, например owner.id на поле дто ownerId. 

Тогда у тебя будет полная дтошечка

drewbal commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public void getOwnedEvents(long userId) {
+        eventRepository.findAllByUserId(userId);
```
Вот они, пустые методы. Красочные и бесполезные как фейерверки :)) запускаются, что-то ищут и достают в базе данных. И все :)

По-хорошему ты получил сейчас список сущностей из БД, а теперь осталось его смаппить в дто и вернуть в контроллер пользователю.

Для маппинга списка используй метом в маппере типа 
List<EventDto> toDtoList(List<Event>)
Он сам сделает маппинг листа по правилам, которые ты определил для одиночных случаев.

drewbal commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public void getParticipatedEvents(long userId) {
+        eventRepository.findParticipatedEventsByUserId(userId);
```
и тут тоже. Почти сделал. На 97%. Осталась капелька, верни данные юзеру и можно веселиться :)

drewbal reviewed (commented):
Хорошая работа, большая. Насчет фильтров думаю можно улучшить, но пока работает, а для масштабирования надо будет подумать.
От меня пара комментов. Все фигня, только закончи последние 2 метода, чтобы они работали как надо.
В остальном гуд, в тестах ты прям постарался, даже какие-то чудные аннотации Мокито применил. Мощь

DeemaP commented on src/main/java/school/faang/user_service/filter/EventFilter.java:
```diff
@@ -0,0 +1,34 @@
+@Component
+public class EventFilter {
+    public List<Event> filterEvents(List<Event> events, EventFilterDto filter) {
+        return events.stream()
```
Объективно эта конструкция выглядит слегка ужасающе О_О, плюс вряд ли удобно будет расширять функционал в дальнейшем, если придется добавить еще какой-то фильтр) 

В лекции есть часть, где рассказывается о фильтрах)
Все эти лишние проверки на null и фильтрацию фильтров по применяемости по-хорошему надо разбить на отдельные классы, которые в себя имплементируют общий интерфейс filter, у которого есть два метода
isApplicable (который проверяет применим ли фильтр) и метод apply, который применяет к стриму твоих сущностей(в данном случае ивентам) каждый конкретный фильтр, и возвращает из него уже отфлитрованный стрим)

Можешь поглядешь у меня в пр как я реализовал. Не уверен на все 100%, что все правильно, но оно работает корректно и выглядит компактнее)


https://github.com/CorporationX/user_service/pull/1791

Я с этими гребаными фильтрами провел больше 12 часов....

DeemaP reviewed (commented):
Еще раз поглядел, пока из очевидного оставил только один коммент))

Darkselect committed:
BJS2-40520: Полностью закончил задачу 7/7

1. Fixed getOwnedEvents method in EventController class

2. Fixed getParticipatedEvents method in EventController class

3. Fixed getOwnedEvents method in EventService class

4. Fixed getParticipatedEvents method in EventService class

All tests are rerited

mighty-mhsl commented on src/main/java/school/faang/user_service/filter/EventFilter.java:
```diff
@@ -0,0 +1,34 @@
+                        .average()
+                        .orElse(0.0) >= filter.getMinRating())
+                .filter(event -> filter.getStatus() == null || event.getStatus() == filter.getStatus())
+                .collect(Collectors.toList());
```
Не не не, это жестко
Завтра у меня новый фильтр - я лезу этот код менять
Нужно сделать по SOLID
Там есть воркшоп, как это делается, на платформе
А еще Дима и Юра у нас в команде уже шарят, как это реализовывать - могут подсказать

Darkselect replied:
Сделаю тогда интерфейс

mighty-mhsl commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    private Event findEventById(long eventId) {
+        return eventRepository.findById(eventId).orElseThrow(() -> new DataValidationException("Event id not found"));
```
Тут скорее EntityNotFoundException, т.к. мы тут не валидируем ничего
Надо мастер подтянуть - там уже есть это исключение

Darkselect replied:
Подтянул

mighty-mhsl commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,72 @@
+
+    public EventDto updateEvent(EventDto eventDto) {
+        Event event = findEventById(eventDto.getId());
+        eventValidation.validateEventDto(eventDto, userService);
```
Мы точно не можем в метод передавать сервис. Либо валидатор должен знать про этот сервис уже, либо мы лишнее делаем. Но параметром точно нельзя сервисы передавать

mighty-mhsl commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,72 @@
+    public EventDto updateEvent(EventDto eventDto) {
+        Event event = findEventById(eventDto.getId());
+        eventValidation.validateEventDto(eventDto, userService);
+        Event updatedEvent = eventMapper.dtoToEvent(eventDto);
```
В разборе по MapStruct я в конце рассказываю, как делать обновление красиво

mighty-mhsl commented on src/main/java/school/faang/user_service/service/user/UserService.java:
```diff
@@ -0,0 +1,18 @@
+public class UserService {
+    private final UserRepository userRepository;
+
+    public User findOwnerById(EventDto event) {
```
Не может UserService знать про эвент. Этот метод должен вообще не знать про другие сущности. Он должен просто id юзера принимать и все
И называться findById. Тогда он будет работать с любыми id, а не только с эвентами. А то нам придется новый точно такой же метод делать каждый раз, когда нам юзера надо получить

mighty-mhsl commented on src/main/java/school/faang/user_service/validation/EventValidation.java:
```diff
@@ -0,0 +1,18 @@
+import school.faang.user_service.exception.DataValidationException;
+
+@Component
+public class EventValidation {
```
EventValidator

mighty-mhsl commented on src/main/java/school/faang/user_service/validation/event/EventServiceValidation.java:
```diff
@@ -0,0 +1,5 @@
+package school.faang.user_service.validation.event;
+
+public class EventServiceValidation {
```
Это надо удалить

mighty-mhsl commented on src/main/java/school/faang/user_service/validation/event/EventValidation.java:
```diff
@@ -0,0 +1,32 @@
+import java.util.List;
+
+@Component
+public class EventValidation {
```
А почему у нас два класса с одинаковым названием? Должен быть один, который всю логику такую содержит

mighty-mhsl commented on src/main/java/school/faang/user_service/validation/event/EventValidation.java:
```diff
@@ -0,0 +1,32 @@
+
+    public void validateEventDto(EventDto event, UserService userService) {
+        validateEvent(event);
+        List<Long> skillsId = userService.findOwnerById(event).getSkills().stream()
```
Это должно быть внутри сервиса, который дергает этот метод validateEventDto. И сюда на валидацию уже передается список скиллов параметром, чтобы валидацию провести
Нельзя передавать зависимости классов параметрами в их методы

mighty-mhsl commented on src/test/java/school/faang/user_service/controller/EventControllerTest.java:
```diff
@@ -0,0 +1,77 @@
+        when(eventService.create(eventDto)).thenReturn(eventDto);
+        eventController.create(eventDto);
+
+        verify(eventService, times(1)).create(eventDto);
```
Надо сделать ассерт возвращаемого значения еще

mighty-mhsl commented on src/test/java/school/faang/user_service/controller/EventControllerTest.java:
```diff
@@ -0,0 +1,77 @@
+
+        when(eventService.getEvent(anyLong())).thenReturn(eventDto);
+        eventController.getEvent(1L);
+        verify(eventService, times(1)).getEvent(anyLong());
```
А ассерт?

mighty-mhsl commented on src/test/java/school/faang/user_service/filter/EventFilterTest.java:
```diff
@@ -0,0 +1,286 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(MockitoExtension.class)
+public class EventFilterTest {
```
Перепишется на отдельные классы для отдельных классов-фильтров

mighty-mhsl commented on src/test/java/school/faang/user_service/service/EventServiceTest.java:
```diff
@@ -0,0 +1,132 @@
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
```
А это зачем?

Darkselect replied:
Уже аргументировал

mighty-mhsl commented on src/test/java/school/faang/user_service/service/EventServiceTest.java:
```diff
@@ -0,0 +1,132 @@
+        verify(eventRepository).findById(eventDto.getId());
+        verify(eventMapper).dtoToEvent(eventDto);
+        verify(eventRepository).save(event);
+        verify(eventMapper).eventToDto(event);
```
Тут не обязательно прям все вызовы проверять. Если ты ассерт сделал, то мапперы де-факто отработали точно
Достаточно вызовы репозиториев проверить
И валидатор точно отработал, раз ты исключение не получил

mighty-mhsl commented on src/test/java/school/faang/user_service/service/event/EventServiceTest.java:
```diff
@@ -0,0 +1,209 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
```
Почему?

Darkselect replied:
А если  мы не ставим эту аннотацию, то все тесты падают. Мне про эту аннотацию подсказали ребята из 6 потока

RuslanZinovyev replied:
Тебе нужно понять как будущему разработчику почему тесты падают, а не использовать аннотации и конфигурации значение которых ты не понимаешь сам, только потому что так сказали ребята из другого потока. На реальной работе ты должен быть способен объяснить каждое свое решение и при случае обосновать его.

RuslanZinovyev replied:
Не понимаю что конкретно тут было исправлено. 
@ExtendWith(MockitoExtension.class) нужная аннотация и удалять ее не надо

mighty-mhsl commented on src/test/java/school/faang/user_service/validation/event/EventValidationTest.java:
```diff
@@ -0,0 +1,91 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
```
Зачем?

Darkselect replied:
Без аннотации не работает

RuslanZinovyev replied:
Ты разобрался почему не работает? Если не можешь понять почему попроси помощи у ребят, нужно разобраться, именно такие проблемы помогают расти и дают опыт. Только сталкиваясь с проблемами мы учимся.

Darkselect replied:
Да, и теперь только эту аннотацию буду использовать: 

@MockitoSettings(strictness = Strictness.STRICT_STUBS):

Этот вариант устанавливает режим строгих заглушек (Strict Stubs), который указывает Mockito выбрасывать исключение, если вы настроили заглушку (stub) для метода, но не вызвали её в тесте.

Это полезно для отлова ошибок, когда некоторые заглушки создаются, но не используются, что может указывать на проблемы в логике теста или избыточные настройки.

Обычно применяют в тестах, когда требуется убедиться, что все заглушки используются корректно, или когда важно быть уверенным в строгости тестов.

Darkselect replied:
При необходимости конечо :)

mighty-mhsl commented on src/test/java/school/faang/user_service/validation/event/EventValidationTest.java:
```diff
@@ -0,0 +1,91 @@
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+public class EventValidationTest {
+    @InjectMocks
+    EventValidation eventValidation;
```
private

Darkselect committed:
BJS2-40520: Полностью закончил задачу 7/7

Fixed filter method
Fixed validation

All tests are rerited

RuslanZinovyev commented on src/main/java/school/faang/user_service/controller/EventController.java:
```diff
@@ -0,0 +1,47 @@
+
+import java.util.List;
+
+@Controller
```
Учитывая что мы будем разрабатывать REST приложение укажи RestController аннотацию, на данном этапе возможно ты понимаешь разницу, мы к этому придем.

RuslanZinovyev commented on src/main/java/school/faang/user_service/controller/EventController.java:
```diff
@@ -0,0 +1,47 @@
+@Slf4j
+public class EventController {
+    private final EventService eventService;
+    private final EventRepository eventRepository;
```
Зависимости EventRepository не должно быть в контроллере, тут мы работаем исключительно с сервисом EventService, сам же сервис использует зависимость EventRepostiory для взаимодействия с базой данных.

RuslanZinovyev commented on src/main/java/school/faang/user_service/validation/event/EventValidation.java:
```diff
@@ -0,0 +1,26 @@
+        }
+    }
+
+    public void validateEventDto(EventDto event, List<Long> skillsId) {
```
Не похоже что данный метод валидирует EventDto, согласно логике метода более уместно было бы имя validateRelatedSkills() вместо validateEventDto()

RuslanZinovyev commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,82 @@
+                .map(Skill::getId)
+                .toList();
+
+        eventValidation.validateEventDto(eventDto, skillsId);
```
Согласно условию задачи метод create должен валидировать EventDto, я не вижу этой валидации, ты используешь метод validateEventDto который по сути валидирует скиллы пользователя, пересмотри внимательно логику валидации.

RuslanZinovyev commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,82 @@
+                .toList();
+
+        eventValidation.validateEventDto(eventDto, skillsId);
+        Event event = eventRepository.save(eventMapper.dtoToEvent(eventDto));
```
Учитывая что eventDto не был провалидирован, конечный результат может быть неожиданным.

RuslanZinovyev commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,82 @@
+        return eventRepository.findById(eventId).orElseThrow(() -> new EntityNotFoundException("Event id not found"));
+    }
+
+    public EventDto getEvent(long eventId) {
```
метод называется getEvent() где Event - это модель, но возвращаешь ты EventDto, переименуй метод тогда в getEventDto()

Darkselect replied:
Внутри метода нужно вызвать метод getEvent(long eventId) класса EventService. 

Я по условию делал

Darkselect replied:
ок сделаю так как ты сказал :)

RuslanZinovyev commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,82 @@
+        return eventMapper.eventToDto(event);
+    }
+
+    private Event findEventById(long eventId) {
```
Хорошей практикой является убирать приватные вспомогательные методы вниз класса.

RuslanZinovyev commented on src/main/java/school/faang/user_service/service/event/EventService.java:
```diff
@@ -0,0 +1,82 @@
+                .map(Skill::getId)
+                .toList();
+
+        eventValidation.validateEventDto(eventDto, skillsId);
```
Имя метода validateEventDto не соотвествует его имплементации.
Cогласно условию задачи, здесь нужно проверить, что пользователь, который создает событие, имеет навыки, связанные с событием. Пересмотри внимательно все имена методов, именование методов очень важная задача ибо по ним другой разработчки должен интуитивно понять что метод делает.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventDescriptionFilterTest.java:
```diff
@@ -0,0 +1,68 @@
+    public void EventDescriptionNotNullTest() {
+        EventFilterDto eventFilterDto = EventFilterDto
+                .builder()
+                .description("Description").
```
точка должна идти впереди вызова, ошибка форматирования

Darkselect committed:
BJS2-40520: Updates

Fixed create method in EventService class
Fixed updateEvent method in EventService class
Fixed validate method in EventValidation class

All tests have been fixed

Darkselect committed:
BJS2-40520: Updates

Added annotations to the classes.

Darkselect committed:
Merge branch 'Dragon-master-stream7' of https://github.com/CorporationX/user_service into Dragon-master-stream7-feature-BJS2-40520-chapter1

# Conflicts:
#	src/main/java/school/faang/user_service/service/event/EventService.java
#	src/test/java/school/faang/user_service/service/event/EventServiceTest.java

Darkselect committed:
BJS2-40520: Updates

Added annotations to the classes.

Darkselect committed:
BJS2-40520: Updates

Added new test for class EventService class

Darkselect committed:
BJS2-40520: Updates

Added new test for class EventService class

Darkselect committed:
BJS2-40520: Updates

Added new test for class EventService class

Darkselect committed:
BJS2-40520: Updates 3.0

Added new test for class EventService class

RuslanZinovyev commented on src/main/java/school/faang/user_service/dto/UserDto.java:
```diff
@@ -28,4 +30,10 @@ public class UserDto {
     private List<Long> menteesId;
     private List<Long> mentorsId;
     private List<Long> skillsId;
+//    private String phone;
```
Любой закоменченный код не должен улетать в мастер, так что либо удали совсем либо оставь.

Darkselect committed:
BJS2-40520: Updates 4.0

Done

RuslanZinovyev commented on src/test/java/school/faang/user_service/controller/EventControllerTest.java:
```diff
@@ -0,0 +1,89 @@
+
+        EventDto result = eventController.create(eventDto);
+
+        assertEquals(expectedEventDto, result);
```
хорошая практика убирать все assertions в самый низ теста

RuslanZinovyev commented on src/main/java/school/faang/user_service/validation/event/EventValidation.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.List;
+
+@Component
+public class EventValidation {
```
EventValidator более подходящее имя для подобного рода классов, имя класса это всегда существительное.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventDescriptionFilterTest.java:
```diff
@@ -0,0 +1,68 @@
+@ExtendWith(MockitoExtension.class)
+public class EventDescriptionFilterTest {
+    @InjectMocks
+    EventDescriptionFilter eventDescriptionFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventEndDateFromFilterTest.java:
```diff
@@ -0,0 +1,70 @@
+
+@ExtendWith(MockitoExtension.class)
+public class EventEndDateFromFilterTest {
+    @InjectMocks
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventIdFilterTest.java:
```diff
@@ -0,0 +1,66 @@
+@ExtendWith(MockitoExtension.class)
+public class EventIdFilterTest {
+   @InjectMocks
+   EventIdFilter eventIdFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventLocationFilterTest.java:
```diff
@@ -0,0 +1,66 @@
+
+@ExtendWith(MockitoExtension.class)
+public class EventLocationFilterTest {
+    @InjectMocks
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventMaxAttendeesFilterTest.java:
```diff
@@ -0,0 +1,64 @@
+
+@ExtendWith(MockitoExtension.class)
+public class EventMaxAttendeesFilterTest {
+    @InjectMocks
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventMinRatingTest.java:
```diff
@@ -0,0 +1,86 @@
+
+@ExtendWith(MockitoExtension.class)
+public class EventMinRatingTest {
+    @InjectMocks
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventOwnerFilterTest.java:
```diff
@@ -0,0 +1,77 @@
+@ExtendWith(MockitoExtension.class)
+public class EventOwnerFilterTest {
+    @InjectMocks
+    EventOwnerIdFilter eventOwnerIdFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventSkillsIdFilterTest.java:
```diff
@@ -0,0 +1,87 @@
+@ExtendWith(MockitoExtension.class)
+public class EventSkillsIdFilterTest {
+    @InjectMocks
+    EventSkillsIdFilter eventSkillsIdFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventStartDateFromTest.java:
```diff
@@ -0,0 +1,73 @@
+@ExtendWith(MockitoExtension.class)
+public class EventStartDateFromTest {
+    @InjectMocks
+    EventStartDateFromFilter startDateFromFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventTitleFilterTest.java:
```diff
@@ -0,0 +1,68 @@
+@ExtendWith(MockitoExtension.class)
+public class EventTitleFilterTest {
+    @InjectMocks
+    private EventTitleFilter eventTitleFilter;
```
А зачем InjectMocks аннотация на этой зависимости, я не вижу никаких моков в тесте. 
Наличие ExtendWith(MockitExtension) тоже под вопросом, ты не делаешь в этом тесте никаких моков и стабов, просто создай инстанс объекта фильтра и тестируй его.

RuslanZinovyev commented on src/test/java/school/faang/user_service/filter/EventTypeFilterTest.java:
```diff
@@ -0,0 +1,88 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
```
Нужна ли тут данная аннотация, я не вижу использования моков

Darkselect committed:
BJS2-40520: Updates 5.0

All tests passed.

Darkselect committed:
Merge branch 'Dragon-master-stream7' of https://github.com/CorporationX/user_service into Dragon-master-stream7-feature-BJS2-40520-chapter1

# Conflicts:
#	src/main/java/school/faang/user_service/dto/skill/SkillDto.java
#	src/main/java/school/faang/user_service/exception/DataValidationException.java
#	src/main/java/school/faang/user_service/service/UserService.java
#	src/main/java/school/faang/user_service/service/mentorship/MentorshipService.java
#	src/test/java/school/faang/user_service/service/user/UserServiceTest.java

Darkselect committed:
BJS2-40520: Updates 6.0

All tests passed.
